finality-provider/service/app.go

```diff
@@ -363,7 +364,13 @@ func (app *FinalityProviderApp) CreateFinalityProvider(
 	if eotsPk == nil {
 		return nil, fmt.Errorf("eots pk cannot be nil")
 	}
-	pop, err := app.CreatePop(fpAddr, eotsPk)
+
+	var signCtx string
+	if app.fpIns != nil && app.config.ContextSigningHeight > app.fpIns.poller.NextHeight()-1 {
+		signCtx = signingcontext.FpPopContextV0(chainID, signingcontext.AccFinality.String())
```

Comment:
FpPop is handled through `x/btcstaking` module, so you should pass `signingcontext.AccBTCstaking` instead of `signingcontext.AccFinality`